### PR TITLE
Add update script to Pocketbase

### PR DESCRIPTION
**Pull request by dsiebel**
> [!NOTE]
> We are meticulous when it comes to merging code into the main branch, so please understand that we may reject pull requests that do not meet the project's standards. It's never personal. Also, game-related scripts have a lower chance of being merged.

## Description

Pocketbase LXC was missing `update_script`, so I added it.

## Type of change
Please check the relevant option(s):

- [ ] Bug fix (non-breaking change that resolves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [ ] New script (a fully functional and thoroughly tested script or set of scripts.)

## Prerequisites
The following efforts must be made for the PR to be considered. Please check when completed:
- [x] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [x] Testing performed (I have tested my changes, ensuring everything works as expected)
- [ ] Documentation updated (I have updated any relevant documentation)

### CODE REVIEW

File: install/pocketbase-install.sh [hunk 1]
```diff
@@ -51,4 +52,5 @@ customize
 msg_info "Cleaning up"
 $STD apt-get -y autoremove
 $STD apt-get -y autoclean
+rm -rf /tmp/pocketbase.zip
```
Comment (havardthom) [inline install/pocketbase-install.sh:55]: move above `autoremove`

File: ct/pocketbase.sh [hunk 2]
```diff
@@ -52,6 +52,36 @@ function default_settings() {
...
+  if [[ ! -f /etc/systemd/system/pocketbase.service ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
+
+  RELEASE=$(curl -s https://api.github.com/repos/pocketbase/pocketbase/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
+
+  if [[ ! -f /opt/${APP}_version.txt || "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]]; then
+    msg_info "Stopping ${APP}"
+    systemctl stop pocketbase.service
+    msg_ok "Stopped ${APP}"
+
+    msg_info "Updating $APP to ${RELEASE}"
```
Comment (havardthom) [inline ct/pocketbase.sh:68]:
```suggestion
    msg_info "Updating $APP to v${RELEASE}"
```

File: ct/pocketbase.sh [hunk 3]
```diff
@@ -52,6 +52,36 @@ function default_settings() {
...
+    wget -q https://github.com/pocketbase/pocketbase/releases/download/v${RELEASE}/pocketbase_${RELEASE}_linux_amd64.zip -O /tmp/pocketbase.zip
+    mkdir -p /opt/pocketbase/{pb_public,pb_migrations,pb_hooks}
+    unzip -q -o /tmp/pocketbase.zip -d /opt/pocketbase
+    echo "${RELEASE}" >/opt/${APP}_version.txt
+    msg_ok "Updated ${APP} to ${RELEASE}"
```
Comment (havardthom) [inline ct/pocketbase.sh:73]:
```suggestion
    msg_ok "Updated ${APP} to v${RELEASE}"
```

File: ct/pocketbase.sh [hunk 4]
```diff
@@ -52,6 +52,36 @@ function default_settings() {
...
+
+    msg_info "Starting ${APP}"
+    systemctl start pocketbase.service
+    msg_ok "Started ${APP}"
+    msg_ok "Updated Successfully"
+  else
+    msg_ok "No update required. ${APP} is already at ${RELEASE}"
```
Comment (havardthom) [inline ct/pocketbase.sh:80]:
```suggestion
    msg_ok "No update required. ${APP} is already at v${RELEASE}"
```

File: ct/pocketbase.sh [hunk 5]
```diff
@@ -52,6 +52,36 @@ function default_settings() {
...
+  RELEASE=$(curl -s https://api.github.com/repos/pocketbase/pocketbase/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
+
+  if [[ ! -f /opt/${APP}_version.txt || "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]]; then
+    msg_info "Stopping ${APP}"
+    systemctl stop pocketbase.service
+    msg_ok "Stopped ${APP}"
+
+    msg_info "Updating $APP to ${RELEASE}"
+    wget -q https://github.com/pocketbase/pocketbase/releases/download/v${RELEASE}/pocketbase_${RELEASE}_linux_amd64.zip -O /tmp/pocketbase.zip
+    mkdir -p /opt/pocketbase/{pb_public,pb_migrations,pb_hooks}
```
Comment (havardthom) [inline ct/pocketbase.sh:70]: Is this needed? directories should already exist from install script

File: ct/pocketbase.sh [hunk 6]
```diff
@@ -52,6 +52,36 @@ function default_settings() {
...
+    unzip -q -o /tmp/pocketbase.zip -d /opt/pocketbase
```
Comment (havardthom) [inline ct/pocketbase.sh:71]: Make sure no migration steps are needed after replacing the executable

**Review by havardthom — DISMISSED** [review]

**Comment by havardthom** [comment]
Also change `updateable: true` in `pocketbase.json`

File: install/pocketbase-install.sh (hunk 1 above)
Comment (dsiebel) [reply install/pocketbase-install.sh:55]: I'm confused. I checked a few other scripts and they all did it _after_ autoremove.

File: ct/pocketbase.sh (hunk 5 above)
Comment (dsiebel) [reply ct/pocketbase.sh:70]: You're right, probably a copy-paste leftover from the install script.

File: install/pocketbase-install.sh (hunk 1 above)
Comment (dsiebel) [reply install/pocketbase-install.sh]: fixed.

**Comment by dsiebel** [comment]
Thanks for the review!
I addressed most of the comments, but I am not sure what is meant by "migrations" when replacing the binary.

File: ct/pocketbase.sh (hunk 6 above)
Comment (dsiebel) [reply ct/pocketbase.sh:70]: So, I just learned that pocketbase had a `pocketbase update` command that backs up the data directory, updates the binary _and_ runs with `--automigrate` by default.

I will switch the update script to using that, in the hopes that this is what you meant.

Comment (dsiebel) [reply ct/pocketbase.sh]: Done: [`a604a97` (#535)](https://github.com/community-scripts/ProxmoxVE/pull/535/commits/a604a974659f73e7ac61a9979567f97ae70c5d65)

File: ct/pocketbase.sh [hunk 7]
```diff
@@ -52,6 +52,18 @@ function default_settings() {
   echo_default
 }
 
+function update_script() {
+  header_info
+  check_container_storage
+  check_container_resources
+  if [[ ! -f /etc/systemd/system/pocketbase.service || ! -x /opt/pocketbase/pocketbase ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
+	/opt/pocketbase/pocketbase update
```
Comment (havardthom) [inline ct/pocketbase.sh:60]:
```suggestion
msg_info "Stopping ${APP}"
systemctl stop pocketbase
msg_ok "Stopped ${APP}"

msg_info "Updating ${APP}"
/opt/pocketbase/pocketbase update
msg_ok "Updated ${APP}"

msg_info "Starting ${APP}"
systemctl start pocketbase
msg_ok "Started ${APP}"
msg_ok "Updated Successfully"
```

File: install/pocketbase-install.sh [hunk 8]
```diff
@@ -43,12 +43,14 @@ WantedBy = multi-user.target
 EOF
 
 systemctl enable -q --now pocketbase.service
+echo "${RELEASE}" > "/opt/${APPLICATION}_version.txt"
```
Comment (havardthom) [inline install/pocketbase-install.sh:46]:
```suggestion
```

**Review by havardthom — DISMISSED** [review]

File: ct/pocketbase.sh [hunk 9]
```diff
@@ -52,6 +52,29 @@ function default_settings() {
   echo_default
 }
 
+function update_script() {
+  header_info
+  check_container_storage
+  check_container_resources
+  if [[ ! -f /etc/systemd/system/pocketbase.service || ! -x /opt/pocketbase/pocketbase ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
+msg_info "Stopping ${APP}"
```
Comment (havardthom) [inline ct/pocketbase.sh:60]: fix indendation

Comment (havardthom) [reply ct/pocketbase.sh:60]: And remove Restarting section

Comment (havardthom) [reply ct/pocketbase.sh:60]: and test that it still works

Comment (dsiebel) [reply ct/pocketbase.sh:60]: Sorry, looks like something didn't quite work with the code suggestions.

Comment (dsiebel) [reply ct/pocketbase.sh]: Fixed. Tested the script on my pocketbase install.
Seems to work, but I am already on the latest version

image

**Review by havardthom — APPROVED** [review]

**Review by MickLesk — APPROVED** [review]

**Pull request merged**